Geometry/DTGeometryBuilder/plugins/dd4hep/DTGeometryValidate.cc

```diff
@@ -0,0 +1,364 @@
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "Geometry/DTGeometry/interface/DTGeometry.h"
+#include "Geometry/DTGeometry/interface/DTLayer.h"
+#include "Geometry/Records/interface/MuonGeometryRcd.h"
+
+#include "Fireworks/Core/interface/FWGeometry.h"
+
+#include "DataFormats/GeometrySurface/interface/RectangularPlaneBounds.h"
+#include "DataFormats/GeometrySurface/interface/TrapezoidalPlaneBounds.h"
+
+#include <TFile.h>
+#include <TH1.h>
+
+#include <cmath>
+#include <limits>
+#include <string>
+#include <type_traits>
+#include <algorithm>
+
+using namespace std;
+
+template<class T>
+typename enable_if<!numeric_limits<T>::is_integer, bool>::type
+    almost_equal(T x, T y, int ulp)
+{
+    // the machine epsilon has to be scaled to the magnitude of the values used
+    // and multiplied by the desired precision in ULPs (units in the last place)
+    return abs(x-y) <= numeric_limits<T>::epsilon() * abs(x+y) * ulp
+        // unless the result is subnormal
+        || abs(x-y) < numeric_limits<T>::min();
+}
```

Comment:
This function is very generic. Shouldn't it go in `DataFormats/Math`? It looks generally useful, not just for DT geometry. Maybe in a later PR.

Reply:
yes, I agree